firedrake/slate/preconditioners.py

```diff
@@ -69,34 +66,27 @@ def initialize(self, pc):
 
         # Automagically determine which spaces are vector and scalar
         for i, Vi in enumerate(V):
-            if Vi.ufl_element().value_shape():
+            if Vi.ufl_element().sobolev_space().name == "HDiv":
                 self.vidx = i
             else:
+                assert Vi.ufl_element().sobolev_space().name == "L2"
                 self.pidx = i
 
         # Create the space of approximate traces.
-        # TODO: Once extruded and tensor product trace elements
-        # are ready, this logic will be updated.
         W = V[self.vidx]
-        hdiv_family = W.ufl_element().family()
-        if hdiv_family == "Raviart-Thomas":
-            tdegree = W.ufl_element().degree() - 1
-
-        elif hdiv_family == "Brezzi-Douglas-Marini":
+        if W.ufl_element().family() == "Brezzi-Douglas-Marini":
```

Comment:
Update: I've leaving this as it. After our discussion today, I feel it's best to just leave this logic here (subject to editing if necessary). Note that if `BDMCF` existed in FIAT, the logic for BDM applies in the same way.